### PR TITLE
VMClone: fix clone_test with snapshot and restore already exists  

### PR DESCRIPTION
**Pull request by Longchuanzheng**
**What this PR does / why we need it**:

1）Chenge reactor functions return nil obj with AlreadyExists err
2）add snapshotcontent in clone_test

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Hi, reviewers.This pr is focuse on change what reactor functions should return with AlreadyExists err. IMO, this pr make clone_test can find the problems fixed by this https://github.com/kubevirt/kubevirt/pull/10903. 
When I dug deeper, I found that if there is no snapshotcontent in the cache, the "with source VM" test will not execute correctly, so I added snapshotcontent for clone_test.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Longchuanzheng** [comment]
@orelmisan  Hi, I dealt with clone_test here.

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 1]
```diff
@@ -220,6 +225,32 @@ var _ = Describe("Clone", func() {
 		})
 	}
 
+	expectCloneUpdateWithSnapshotName := func(phase clonev1alpha1.VirtualMachineClonePhase, snapshotName string) {
+		client.Fake.PrependReactor("update", clone.ResourceVMClonePlural, func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			update, ok := action.(testing.UpdateAction)
+			Expect(ok).To(BeTrue())
+
+			vmClone := update.GetObject().(*clonev1alpha1.VirtualMachineClone)
```
Comment (0xFelix) [inline pkg/virt-controller/watch/clone/clone_test.go:233]: Check this cast too, as above?

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 2]
```diff
@@ -220,6 +225,32 @@ var _ = Describe("Clone", func() {
...
+			return true, update.GetObject(), nil
+		})
+	}
+
+	expectCloneUpdateWithRestoreName := func(phase clonev1alpha1.VirtualMachineClonePhase, restoreName string) {
+		client.Fake.PrependReactor("update", clone.ResourceVMClonePlural, func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			update, ok := action.(testing.UpdateAction)
+			Expect(ok).To(BeTrue())
+
+			vmClone := update.GetObject().(*clonev1alpha1.VirtualMachineClone)
```
Comment (0xFelix) [inline pkg/virt-controller/watch/clone/clone_test.go:246]: Same as above.

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 3]
```diff
@@ -220,6 +225,32 @@ var _ = Describe("Clone", func() {
...
 	}
 
+	expectCloneUpdateWithSnapshotName := func(phase clonev1alpha1.VirtualMachineClonePhase, snapshotName string) {
+		client.Fake.PrependReactor("update", clone.ResourceVMClonePlural, func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			update, ok := action.(testing.UpdateAction)
+			Expect(ok).To(BeTrue())
+
+			vmClone := update.GetObject().(*clonev1alpha1.VirtualMachineClone)
+			Expect(vmClone.Status.Phase).To(Equal(phase))
+			Expect(*vmClone.Status.SnapshotName).To(Equal(snapshotName))
```
Comment (0xFelix) [inline pkg/virt-controller/watch/clone/clone_test.go:235]: Could use `HaveValue` instead: https://pkg.go.dev/github.com/onsi/gomega#HaveValue

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 4]
```diff
@@ -220,6 +225,32 @@ var _ = Describe("Clone", func() {
...
+	}
+
+	expectCloneUpdateWithRestoreName := func(phase clonev1alpha1.VirtualMachineClonePhase, restoreName string) {
+		client.Fake.PrependReactor("update", clone.ResourceVMClonePlural, func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			update, ok := action.(testing.UpdateAction)
+			Expect(ok).To(BeTrue())
+
+			vmClone := update.GetObject().(*clonev1alpha1.VirtualMachineClone)
+			Expect(vmClone.Status.Phase).To(Equal(phase))
+			Expect(*vmClone.Status.RestoreName).To(Equal(restoreName))
```
Comment (0xFelix) [inline pkg/virt-controller/watch/clone/clone_test.go:248]: Same as above.

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 5]
```diff
@@ -962,6 +996,30 @@ func createVirtualMachineRestore(vm *virtv1.VirtualMachine, snapshotName string)
...
 }
 
+func createVirtualMachineSnapshotSnapshotContent(vm *virtv1.VirtualMachine, snapshotName string) *snapshotv1alpha1.VirtualMachineSnapshotContent {
+	vmCpy := &snapshotv1alpha1.VirtualMachine{}
+	vmCpy.ObjectMeta = *vm.ObjectMeta.DeepCopy()
+	vmCpy.Spec = *vm.Spec.DeepCopy()
+	vmCpy.ResourceVersion = "1"
+	vmCpy.Status = *vm.Status.DeepCopy()
+	return &snapshotv1alpha1.VirtualMachineSnapshotContent{
+
```
Comment (0xFelix) [inline pkg/virt-controller/watch/clone/clone_test.go:1006]: nit: newline

File: pkg/virt-controller/watch/clone/clone_test.go (hunk 1 above)
Comment (orelmisan) [reply pkg/virt-controller/watch/clone/clone_test.go:233]: *Type assertion
https://go.dev/tour/methods/15

Comment (Longchuanzheng) [reply pkg/virt-controller/watch/clone/clone_test.go]: Done. However, could you please explain the reason for needing a check at this point?

File: pkg/virt-controller/watch/clone/clone_test.go (hunk 5 above)
Comment (Longchuanzheng) [reply pkg/virt-controller/watch/clone/clone_test.go:1023]: Is that what you mean?

**Pull request closed without merging**